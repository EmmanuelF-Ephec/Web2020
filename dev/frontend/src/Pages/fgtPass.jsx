import React, { Component } from "react";

class FgtPass extends Component {
  state = {};
  render() {
    return <div id="fgtPass">Password</div>;
  }
}

export default FgtPass;
