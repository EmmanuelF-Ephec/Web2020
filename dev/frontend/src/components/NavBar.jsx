import React, { Component } from "react";
import { Jumbotron } from "frontend";
export default class NavBar extends Component {
  state = {};
  render() {
    return test;
  }
}
