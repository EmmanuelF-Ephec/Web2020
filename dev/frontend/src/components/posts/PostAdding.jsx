import React, { Component } from "react";
import { NavigationBar } from "../NavigationBar";
import { Form, Button } from "react-bootstrap";

class postAdding extends Component {
  state = {};
  handleSubmit = (event) => {
    event.preventDefault();

    if (formValid(this.state.formErrors)) {
      console.log(
        this.state.firstName,
        this.state.lastName,
        this.state.password,
        this.state.email
      );
    } else {
      console.log("Erreur dans le formulaire");
    }
  };
  render() {
    return (
      <div>
        <NavigationBar />
        <Form>
          <Form.Group controlId="postTitle">
            <Form.Label>Titre</Form.Label>
            <Form.Control placeholder="Titre de l'annonce" />
          </Form.Group>
          <Form.Group controlId="postText">
            <Form.Label>Example textarea</Form.Label>
            <Form.Control as="textarea" rows="5" />
          </Form.Group>
          <Button variant="primary" type="submit">
            Submit
          </Button>
        </Form>
      </div>
    );
  }
}

export default postAdding;
