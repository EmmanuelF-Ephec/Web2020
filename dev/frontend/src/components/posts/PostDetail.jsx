import React, { Component } from "react";
import { NavigationBar } from "../NavigationBar";
import { Container, Jumbotron as Jumbo } from "react-bootstrap";

class PostDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {
      id: null,
    };
  }
  componentDidMount() {
    if (this.props.match) {
      const { id } = this.props.match.params;
      this.setState({
        id: id,
      });
    }
  }

  render() {
    const { id } = this.state;
    return (
      <div>
        {id !== null ? (
          <div>
            <NavigationBar />
            <Jumbo fluid className="jumbo">
              <Container>
                <h1>Titre de l'annonce</h1>
                <p>Sous-titres? de cette superbe annonce</p>
              </Container>
            </Jumbo>
            <Container>
              {" "}
              "But I must explain to you how all this mistaken idea of
              denouncing pleasure and praising pain was born and I will give you
              a complete account of the system, and expound the actual teachings
              of the great explorer of the truth, the master-builder of human
              happiness. No one rejects, dislikes, or avoids pleasure itself,
              because it is pleasure, but because those who do not know how to
              pursue pleasure rationally encounter consequences that are
              extremely painful. Nor again is there anyone who loves or pursues
              or desires to obtain pain of itself, because it is pain, but
              because occasionally circumstances occur in which toil and pain
              can procure him some great pleasure. To take a trivial example,
              which of us ever undertakes laborious physical exercise, except to
              obtain some advantage from it? But who has any right to find fault
              with a man who chooses to enjoy a pleasure that has no annoying
              consequences, or one who avoids a pain that produces no resultant
              pleasure?"
            </Container>
          </div>
        ) : (
          "Rien de trouv√©"
        )}
      </div>
    );
  }
}

export default PostDetail;
