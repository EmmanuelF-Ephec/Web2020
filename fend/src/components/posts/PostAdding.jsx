import React, { Component } from "react";
import NavigationBar from "../NavigationBar";
import { Form, Button, Alert, Container } from "react-bootstrap";
import axios from "axios";
const formValid = (formErrors) => {
  let valid = true;

  Object.values(formErrors).forEach((val) => {
    val.length > 0 && (valid = false);
  });

  return valid;
};

class postAdding extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: null,
      text: null,
      idUser: "1",
      dateTime: null,
      show: false,
      formErrors: {
        title: "",
        text: "",
        idUser: "",
        dateTime: "",
      },
    };
  }

  handleSubmit = (event) => {
    event.preventDefault();
    if (formValid(this.state.formErrors)) {
      const post = {
        titre: this.state.title,
        texteannonce: this.state.text,
        idUtil: this.state.idUser,
        dateCreation: this.state.dateTime,
      };
      axios
        .post("/annonces/", JSON.stringify(post), {
          headers: {
            "Content-Type": "application/json",
          },
        })
        .then((res) => {
          console.log(res.data);
        });
    } else {
      console.log("Erreur dans le formulaire");
    }
    this.setState({
      show: true,
    });
  };

  handleChange = (event) => {
    event.preventDefault();

    const { name, value } = event.target;
    this.setState({ [name]: value }, () => console.log(this.state));
  };

  render() {
    return (
      <div>
        <NavigationBar />
        <Container>
          <Form onSubmit={this.handleSubmit}>
            <Form.Group controlId="postTitle" onChange={this.handleChange}>
              <Form.Label>Titre</Form.Label>
              <Form.Control name="title" placeholder="Titre de l'annonce" />
            </Form.Group>
            <Form.Group controlId="postText" onChange={this.handleChange}>
              <Form.Label>Texte de l'annonce</Form.Label>
              <Form.Control name="text" as="textarea" rows="5" />
            </Form.Group>
            <Button variant="primary" type="submit">
              Submit
            </Button>
          </Form>
          <Alert variant="success" show={this.state.show}>
            Annonce post√©e!
          </Alert>
        </Container>
      </div>
    );
  }
}

export default postAdding;
