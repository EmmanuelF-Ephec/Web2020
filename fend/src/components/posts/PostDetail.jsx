import React, { Component } from "react";
import  NavigationBar from "../NavigationBar";
import { Container, Jumbotron as Jumbo } from "react-bootstrap";
const axios = require("axios").default;

class PostDetail extends Component {
  constructor(props) {
    super(props);
    this.state = {
      id: null,
      title: null,
      content: null,
      didLoad: false,
    };
  }
  componentDidMount() {
    let currentComponent = this;
    if (this.props.match) {
      const { id } = this.props.match.params;
      currentComponent.setState({
        id: id,
      });
      axios
        .get(`/notices/${id}/`)
        .then(function (response) {
          // handle success
          console.log("reussite");
          currentComponent.setState({
            title: response.data.title,
            content: response.data.content,
            didLoad: true,
          });
        })
        .catch(function (error) {
          // handle error
          console.log(error);
        });
    }
  }

  render() {
    const { title } = this.state;
    const { content } = this.state;
    const { didLoad } = this.state;

    return (
      <div>
        {didLoad === true ? (
          <div>
            <NavigationBar />
            <Jumbo fluid className="jumbo">
              <Container>
                <h1>{title}</h1>
              </Container>
            </Jumbo>
            <Container>{content}</Container>
          </div>
        ) : (
          <p> Rien de trouv√©</p>
        )}
      </div>
    );
  }
}

export default PostDetail;
