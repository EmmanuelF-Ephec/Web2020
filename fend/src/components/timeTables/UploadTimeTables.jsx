import React, { Component } from "react";
import axios, { post } from "axios";
import { Form, Button } from "react-bootstrap";

class UploadTimeTables extends Component {
  constructor(props) {
    super(props);
    this.state = {
      file: null,
      name: null,
    };
    this.onFormSubmit = this.onFormSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.fileUpload = this.fileUpload.bind(this);
  }
  onFormSubmit(e) {
    e.preventDefault(); // Stop form submit
    console.log(this.state.name);
    this.fileUpload(this.state.file).then((response) => {
      console.log(this.state.name);
      axios.post("/schedule/", {
        name: this.state.name,
        url: response.data,
      });
      console.log(response.data);
    });
    window.location.reload(false);
  }
  onChange(e) {
    if (e.target.name == "name") {
      this.setState({
        name: e.target.value,
      });
    }
    if (null != e.target.files) {
      this.setState({
        file: e.target.files[0],
      });
    }
  }

  fileUpload(file) {
    const url = "http://127.0.0.1:8000/upload/";
    const formData = new FormData();
    formData.append("file", file);
    const config = {
      headers: {
        "content-type": "multipart/form-data",
      },
    };
    return post(url, formData, config);
  }

  render() {
    return (
      <Form onSubmit={this.onFormSubmit}>
        <h1>Ajoutez les horaires ici!</h1>
        <input
          name="name"
          type="text"
          onChange={this.onChange}
          placeholder="Nom de l'horaire ici"
        />
        <p></p>
        <input type="file" onChange={this.onChange} />
        <Button type="submit">Upload</Button>
        <p></p>
      </Form>
    );
  }
}
export default UploadTimeTables;
