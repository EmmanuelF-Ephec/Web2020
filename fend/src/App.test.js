import React from "react";
import { shallow, configure } from "enzyme";
import { render } from "react-dom";
import Adapter from "enzyme-adapter-react-16";

import Login from "./Pages/login";
import TimeTables from "./Pages/timeTables";
import Announcements from "./Pages/announcements";
import fgtPass from "./Pages/fgtPass";
import NavBar from "./components/NavigationBar";
import SideNav from "./components/SideNavigation";
import PostDetail from "./components/posts/PostDetail";
import PostAdding from "./components/posts/PostAdding";
import UploadTimeTables from "./components/timeTables/UploadTimeTables";

configure({ adapter: new Adapter() });

jest.dontMock("bootstrap");

describe("Login", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("TimeTables", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<TimeTables />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe("announcements", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Announcements />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe("fgtPassword", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<fgtPass />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe("Navigation Bar", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe("SideNavigation", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SideNav />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe("Adding a post", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<PostAdding />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Detail of a post", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<PostDetail />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Uploading a time table", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<UploadTimeTables />);
    expect(wrapper).toMatchSnapshot();
  });
});
