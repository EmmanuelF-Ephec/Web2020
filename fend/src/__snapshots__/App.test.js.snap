// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding a post renders correctly 1`] = `ShallowWrapper {}`;

exports[`Detail of a post renders correctly 1`] = `ShallowWrapper {}`;

exports[`Login renders correctly 1`] = `ShallowWrapper {}`;

exports[`SideNavigation renders correctly 1`] = `ShallowWrapper {}`;

exports[`TimeTables renders correctly 1`] = `ShallowWrapper {}`;

exports[`Uploading a time table renders correctly 1`] = `ShallowWrapper {}`;

exports[`announcements renders correctly 1`] = `ShallowWrapper {}`;

exports[`fgtPassword renders correctly 1`] = `ShallowWrapper {}`;

exports[`registration renders correctly 1`] = `ShallowWrapper {}`;
